### PR TITLE
Documentation 2239 user guide

### PR DESCRIPTION
**Pull request by sergiupacurariu**

### CODE REVIEW

File: sormas-backend/doc/UserDataAccess.md [hunk 1]
```diff
@@ -1,62 +1,131 @@
...
+
+## Basics
 In general data access & synchronisation is based on the following rules:
 
 * hospital informant: all data associated to the facility (cases)
 * community informant: all data associated to the community (cases)
 * officers: all data associated to the district (cases, events)
 * supervisors: all data associated to the region
+* observers: all data associated to their jurisdiction (nation, region or district)
+* port of entry users: all data associated to their jurisdiction (nation, region or district)
```
Comment (MateStrysewskeSym) [inline sormas-backend/doc/UserDataAccess.md:11]: Port of entry -> Port health

Also, some point of entry users always have a point of entry assigned, so it's basically them being able to view data associated to their point of entry; only national point of entry users see data from multiple points of entry

File: sormas-backend/doc/UserDataAccess.md [hunk 2]
```diff
@@ -1,62 +1,131 @@
...
 * special case: if an officer has access to a task or contact whose case/event/contact is not available, the association link should be inactive
-* user rights (e.g. allowed to see cases) are documented in the [user rights table](sormas-api/src/main/resources/doc/SORMAS_User_Rights.xlsx)
+* user rights (e.g. allow seeing cases) and mapping of user roles to a jurisdiction level is documented in the [user rights table](https://github.com/hzi-braunschweig/SORMAS-Project/tree/development/sormas-api/src/main/resources/doc/SORMAS_User_Rights.xlsx)
 
-# Case
+## Case
 * only users that are permitted to see cases at all
 * whoever created the case or is assigned to it is allowed to access it
-* access by region/district/community/health facility/point of entry of the case
+* all cases accessible for National users
```
Comment (MateStrysewskeSym) [inline sormas-backend/doc/UserDataAccess.md:21]: All cases are accessible for national users

File: sormas-backend/doc/UserDataAccess.md [hunk 3]
```diff
@@ -1,62 +1,131 @@
...
+* access by jurisdiction, that is based region/district/community/health facility/point of entry of the case
```
Comment (MateStrysewskeSym) [inline sormas-backend/doc/UserDataAccess.md:22]: [...] that is based on the [...]

File: sormas-backend/doc/UserDataAccess.md [hunk 4]
```diff
@@ -1,62 +1,131 @@
...
 * cases that are made public can be accessed by all users permitted to see cases (this feature can be disabled)
+* edit is possible only for cases that are part of the users jurisdiction
```
Comment (MateStrysewskeSym) [inline sormas-backend/doc/UserDataAccess.md:24]: editing [...] user's jurisdiction

File: sormas-backend/doc/UserDataAccess.md [hunk 5]
```diff
@@ -1,62 +1,131 @@
...
+* view of personal and sensitive data is possible only for cases that are part of the user's jurisdiction
```
Comment (MateStrysewskeSym) [inline sormas-backend/doc/UserDataAccess.md:25]: viewing personal [...]

File: sormas-backend/doc/UserDataAccess.md [hunk 6]
```diff
@@ -1,62 +1,131 @@
...
 
-# Contact
+## Contact
 * only users that are permitted to see contacts at all
 * whoever created it or is assigned to it is allowed to access it
+* all contacts accessible for National users
```
Comment (MateStrysewskeSym) [inline sormas-backend/doc/UserDataAccess.md:30]: see above

File: sormas-backend/doc/UserDataAccess.md [hunk 7]
```diff
@@ -1,62 +1,131 @@
...
 * users see all contacts of their cases
-* access by region/district of the contact
+* access by jurisdiction, that is using region/district of the contact and jurisdiction of the case the contact is assigned to
```
Comment (MateStrysewskeSym) [inline sormas-backend/doc/UserDataAccess.md:32]: see above;
also, this should be a bit clearer:

access by jurisdiction that is using the region/district of the contact or, if not available, the jurisdiction of the source case of the contact

File: sormas-backend/doc/UserDataAccess.md [hunk 8]
```diff
@@ -1,62 +1,131 @@
...
+* edit is possible only for contacts that are part of the users jurisdiction
```
Comment (MateStrysewskeSym) [inline sormas-backend/doc/UserDataAccess.md:33]: see above

File: sormas-backend/doc/UserDataAccess.md [hunk 9]
```diff
@@ -1,62 +1,131 @@
...
+* view of personal and sensitive data is possible only for contacts that are part of the user's jurisdiction
```
Comment (MateStrysewskeSym) [inline sormas-backend/doc/UserDataAccess.md:34]: see above

File: sormas-backend/doc/UserDataAccess.md [hunk 10]
```diff
@@ -1,62 +1,131 @@
...
 
-# Visits
+## Visits
 * users see all visits of the user's contacts
+* view of sensitive data is possible only for contacts that are part of the user's jurisdiction
```
Comment (MateStrysewskeSym) [inline sormas-backend/doc/UserDataAccess.md:38]: see above

File: sormas-backend/doc/UserDataAccess.md [hunk 11]
```diff
@@ -1,62 +1,131 @@
...
 
-# Event
+## Event
 * only users that are permitted to see events at all
 * whoever created the event or is assigned to it is allowed to access it
+* all events accessible for National users
```
Comment (MateStrysewskeSym) [inline sormas-backend/doc/UserDataAccess.md:43]: see above

File: sormas-backend/doc/UserDataAccess.md [hunk 12]
```diff
@@ -1,62 +1,131 @@
...
 * access by region/district of the event
 
-# EventParticipant
+## EventParticipant
 * users see all participants of all events they can access
+* view of personal and sensitive data is possible only for events that are part of the user's jurisdiction
```
Comment (MateStrysewskeSym) [inline sormas-backend/doc/UserDataAccess.md:48]: see above

File: sormas-backend/doc/UserDataAccess.md [hunk 13]
```diff
@@ -1,62 +1,131 @@
...
-# Tasks
+## Tasks
 * only users that are permitted to see tasks at all
 * whoever created the task or is assigned to it is allowed to access it
-* all tasks for the user's cases
-* all tasks for the user's contacts
-* all tasks for the user's events
-
-# Samples
+* all tasks accessible for National users
```
Comment (MateStrysewskeSym) [inline sormas-backend/doc/UserDataAccess.md:53]: see above

File: sormas-backend/doc/UserDataAccess.md [hunk 14]
```diff
@@ -1,62 +1,131 @@
...
+* all tasks for the user's cases except the ones assigned to users in other jurisdiction
+* all tasks for the user's contacts except the ones assigned to users in other jurisdiction
+* all tasks for the user's events except the ones assigned to users in other jurisdiction
+ 
+## Samples
 * only users that are permitted to see samples at all
+* all samples accessible for National Users
```
Comment (MateStrysewskeSym) [inline sormas-backend/doc/UserDataAccess.md:60]: see above

File: sormas-backend/doc/UserDataAccess.md [hunk 15]
```diff
@@ -1,62 +1,131 @@
...
 * whoever created the sample or is assigned to it is allowed to access it
 * users see all samples of their cases
 * lab users see all samples of their laboratory
+* access by jurisdiction of the case or contact the sample belongs to
+* edit is possible only for samples of cases and/or contacts in the user's jurisdiction
```
Comment (MateStrysewskeSym) [inline sormas-backend/doc/UserDataAccess.md:65]: Samples can now also be added for event participants, so this should be reflected here

File: sormas-backend/doc/UserDataAccess.md [hunk 16]
```diff
@@ -1,62 +1,131 @@
...
+* view of sensitive data is possible only for samples that are part of the user's jurisdiction
 
-# Persons
+## Persons
 * all persons of the cases the user can access
 * all persons of the contacts the user can access
 * all persons of the events the user can access
+* edit is possible only only for persons that have their corresponding contact, case or event in the users jurisdiction
+* view of personal and sensitive data is possible only for persons that have their corresponding contact, case or event in the users jurisdiction
```
Comment (MateStrysewskeSym) [inline sormas-backend/doc/UserDataAccess.md:73]: see above

File: sormas-backend/doc/UserDataAccess.md [hunk 17]
```diff
@@ -1,62 +1,131 @@
...
+ 
+## Samples
 * only users that are permitted to see samples at all
+* all samples accessible for National Users
 * whoever created the sample or is assigned to it is allowed to access it
 * users see all samples of their cases
 * lab users see all samples of their laboratory
+* access by jurisdiction of the case or contact the sample belongs to
+* edit is possible only for samples of cases and/or contacts in the user's jurisdiction
+* view of sensitive data is possible only for samples that are part of the user's jurisdiction
```
Comment (MateStrysewskeSym) [inline sormas-backend/doc/UserDataAccess.md:66]: see above

File: sormas-backend/doc/UserDataAccess.md [hunk 18]
```diff
@@ -1,62 +1,131 @@
...
+* When: task of type contact follow-up set to status done
+
+## New and pending task notification
+* Types: email, sms
+* To: [assigned user](#task-generation-and-user-assignment) of each new or pending task
+* When: every 10 minutes 
+
+## Visit changed notification
+* Types: email, sms
+* To: surveillance and contact supervisor of visits contacts region
```
Comment (MateStrysewskeSym) [inline sormas-backend/doc/UserDataAccess.md:116]: [...] of the visit's contact's region

File: sormas-backend/doc/UserDataAccess.md [hunk 19]
```diff
@@ -1,62 +1,131 @@
...
+* When: visit having contacts becomes symptomatic
+
+# Task generation and user assignment
+
+## Submit weekly reports
+* generated every hour if WEEKLY_REPORT_GENERATION feature enabled for each hospital informant
```
Comment (MateStrysewskeSym) [inline sormas-backend/doc/UserDataAccess.md:122]: Only if no weekly report has yet been sent

File: sormas-backend/doc/UserDataAccess.md [hunk 20]
```diff
@@ -1,62 +1,131 @@
...
+
+## Contact follow-up
+* generated every hour if CONTACT_FOLLOW_UP feature enabled 
+* assigned user becomes the contact officer if exists, if not a random officer of the contact district
```
Comment (MateStrysewskeSym) [inline sormas-backend/doc/UserDataAccess.md:126]: I don't really get what this is supposed to mean

File: sormas-backend/doc/UserDataAccess.md [hunk 21]
```diff
@@ -1,62 +1,131 @@
...
+
+## Case related tasks
+* assigned user becomes the surveillance officer of the case if exists, if not a random surveillance officer of the district
```
Comment (MateStrysewskeSym) [inline sormas-backend/doc/UserDataAccess.md:129]: Same as above

File: sormas-backend/doc/UserDataAccess.md (hunk 20 above)
Comment (BarnaBartha) [reply sormas-backend/doc/UserDataAccess.md:126]: it is a description of the logic how the assignee is choosen in the ContactFacadeEjb.generateContactFollowUpTasks method - thought it's relevant to whom a task get's assigned and that's why included it here
should i reformulate it ?

File: sormas-backend/doc/UserDataAccess.md (hunk 1 above)
Comment (BarnaBartha) [reply sormas-backend/doc/UserDataAccess.md:11]: please check this again, how i extended

File: sormas-backend/doc/UserDataAccess.md (hunk 20 above)
Comment (MateStrysewskeSym) [reply sormas-backend/doc/UserDataAccess.md:126]: Ah, okay, I see. I'd rephrase it as follows:

"the task assignee is set to the contact officer if specified, otherwise to a random officer in the contact district"

And basically the same concept for the open discussion point below.
